packages/backend-app-api/src/config/config.ts

```diff
@@ -29,16 +29,19 @@ import type { Config, AppConfig } from '@backstage/config';
 import { getPackages } from '@manypkg/get-packages';
 import { ObservableConfigProxy } from './ObservableConfigProxy';
 import { isValidUrl } from '../lib/urls';
+import { JsonObject } from '@backstage/types';
 
 /** @public */
 export async function createConfigSecretEnumerator(options: {
   logger: LoggerService;
   dir?: string;
+  additionalSchemas?: { [context: string]: JsonObject };
 }): Promise<(config: Config) => Iterable<string>> {
   const { logger, dir = process.cwd() } = options;
   const { packages } = await getPackages(dir);
   const schema = await loadConfigSchema({
     dependencies: packages.map(p => p.packageJson.name),
+    additionalSchemas: options.additionalSchemas,
   });
```

Comment:
This is a key place where I think we could introduce an abstraction that should make you able to set this up as needed. I'm thinking we could add an optional `schema` option to `createConfigSecretEnumerator`, `schema?: ConfigSchema`? If a schema is provided it simply won't to try to load it itself.

On top of that I think the `packagePaths` option for `loadConfigSchema` might actually be enough for you load load in additional schemas btw? But if it's not we can add the `additionalSchemas` option or similar.

Reply:
Done.
I even removed the `additionalSchemas` option.